packages/next-on-pages/templates/_worker.js/utils/cache.ts

```diff
@@ -71,6 +72,29 @@ export async function handleSuspenseCacheRequest(request: Request) {
  * @returns Adaptor for the suspense cache.
  */
 export async function getSuspenseCacheAdaptor(): Promise<CacheAdaptor> {
-	// TODO: Try to lazy import the custom cache adaptor.
-	return new CacheApiAdaptor();
+	if (process.env.KV_SUSPENSE_CACHE) {
+		return getInternalCacheAdaptor('kv');
+	}
+
+	return getInternalCacheAdaptor('cache-api');
+}
+
+type InternalAdaptors = 'kv' | 'cache-api';
+type DeriveAdaptorType<T extends InternalAdaptors> = T extends 'kv'
```

Comment:
Is there a benefit in having different types of adaptors? couldn't/shouldn't all the adaptors just follow a single Adaptor type contract/interface?

Reply:
my bad, these both extend CacheAdaptor so i can just use that 🤦🏻‍♂️